runners/spark/src/main/java/org/apache/beam/runners/spark/SparkRunner.java

```diff
@@ -324,7 +327,7 @@ public void visitPrimitiveTransform(TransformHierarchy.Node node) {
       LOG.info("Evaluating {}", transform);
       AppliedPTransform<?, ?, ?> appliedTransform = node.toAppliedPTransform();
       ctxt.setCurrentTransform(appliedTransform);
-      evaluator.evaluate(transform, ctxt);
+      evaluator.evaluate(transform, ctxt, pview);
```

Comment:
Why pass the `SparkPCollectionView` when the `PipelineVisitor` traverses ? this only happens once, on first run. I'm also pretty sure this never happens again (such as on resume from checkpoint)